yield_plugin/include/yield_plugin/yield_plugin.hpp

```diff
@@ -252,6 +264,16 @@ class YieldPlugin
    */
   std::optional<rclcpp::Time> get_collision_time(const carma_planning_msgs::msg::TrajectoryPlan& original_tp, const carma_perception_msgs::msg::ExternalObject& curr_obstacle);
 
+  /**
+   * \brief Check if object location is behind the vehicle using estimates of the vehicle's length and route downtracks
+   * \param object_id object id to use for the consecutive_clearance_count_for_obstacles_
+   * \param collision_data info about collision such as collision_time, vehicle and obstacle positions at the time of collision
+   * NOTE: Uses internal counter low pass filter to confirm the object is behind only if it counted continuously above
+           config_.consecutive_clearance_count_for_obstacles_threshold
+   * \return return true if object is behind the vehicle
+   */
+  bool is_object_behind_the_vehicle(uint32_t object_id, const CollisionData& collision_data);
```

Comment:
Consider changing this function's signature to use `lanelet::BasicPoint2d` values directly. This function does not need to be coupled to `get_collision_data()` calls.

Reply:
Extracted to use the individual members of collision result